deployer/src/proxy.rs

```diff
@@ -36,41 +40,27 @@ pub async fn handle(
     });
     span.set_parent(parent_context);
 
-    let host: FQDN = match req.headers().get(HOST) {
-        Some(host) => host
-            .to_str()
-            .unwrap_or_default()
-            .parse::<FQDN>()
-            .unwrap_or_default()
-            .to_owned(),
-        None => {
-            trace!("proxy request has no host header");
+    // First try to get the Host header. This header is not allowed in H2, so if we
+    // can't find it we try to get the host subcomponent from the URI authority.
+    let Some(host) = req
+        .headers()
+        .typed_get::<Host>()
+        .map(|host| fqdn!(host.hostname()))
+        .or_else(|| req.uri().host().map(|host| fqdn!(host))) else {
```

Comment:
This is a common way to get the host from http1 | http2, actix-web uses a similar strategy: https://docs.rs/actix-web/4.3.1/src/actix_web/guard/host.rs.html#58-63